dynatrace/stackstate_checks/dynatrace/dynatrace.py

```diff
@@ -0,0 +1,507 @@
+        """
+        labels = []
+        # append management zones in labels for each existing component
+        for zone in dynatrace_component.managementZones:
+            if zone.get("name"):
+                labels.append("managementZones:%s" % zone.get("name"))
+        if dynatrace_component.entityId:
+            labels.append(dynatrace_component.entityId)
+        if dynatrace_component.monitoringState:
+            if dynatrace_component.monitoringState.actualMonitoringState:
+                labels.append("actualMonitoringState:%s" % dynatrace_component.monitoringState.actualMonitoringState)
+            if dynatrace_component.monitoringState.expectedMonitoringState:
+                labels.append(
+                    "expectedMonitoringState:%s" % dynatrace_component.monitoringState.expectedMonitoringState)
+        for technologies in dynatrace_component.softwareTechnologies:
+            tech_label = ''
+            if technologies.get('type'):
+                tech_label += technologies['type']
+            if technologies.get('edition'):
+                tech_label += ":%s" % technologies['edition']
+            if technologies.get('version'):
+                tech_label += ":%s" % technologies['version']
+            labels.append(tech_label)
+        labels_from_tags = self._get_labels_from_dynatrace_tags(dynatrace_component)
+        labels.extend(labels_from_tags)
+        # prefix the labels with `dynatrace-` for all labels
+        labels = ["dynatrace-%s" % label for label in labels]
+        return labels
+
+    def _process_events(self, instance_info):
+        """
+        Wrapper to collect events, filters those events and persist the state
+        """
+        entities_with_events = []
+        events, events_limit_reached = self._collect_events(instance_info)
+        closed_events = len([e for e in events if e.get('eventStatus') == 'CLOSED'])
+        open_events = len([e for e in events if e.get('eventStatus') == 'OPEN'])
+        self.log.info("Collected %d events, %d are open and %d are closed.", len(events), open_events, closed_events)
+        for event in events:
+            self._create_event(event, instance_info.url)
+            entities_with_events.append(event.entityId)
+        # Simulating OK health state by sending CLOSED events for processed topology entities with no events.
+        simulated_count = 0
+        for entity_id in [e for e in dynatrace_entities_cache.keys() if e not in entities_with_events]:
+            simulated_closed_event = DynatraceEvent(
+                {
+                    "eventId": -1,
+                    "startTime": self._current_time_seconds() * 1000,
+                    "endTime": self._current_time_seconds() * 1000,
+                    "entityId": entity_id,
+                    "entityName": dynatrace_entities_cache[entity_id].get('name'),
+                    "impactLevel": None,
+                    "eventType": "CUSTOM_ALERT",
```

Comment:
may be a good name - `DEFAULT HEALTH`. OK health is not a custom alert I think but more a default health which you are sending for all entities who have no events.

Reply:
Changed to DEFAULT HEALTH

Reply:
Done